packages/app/src/app/overmind/effects/api/apiFactory.ts

```diff
@@ -1,85 +1,59 @@
-import axios, { AxiosResponse, AxiosError } from 'axios';
+import axios, { AxiosResponse, AxiosError, AxiosRequestConfig } from 'axios';
 import { logError } from '@codesandbox/common/lib/utils/analytics';
 import { values } from 'lodash-es';
 import { camelizeKeys, decamelizeKeys } from 'humps';
+import { Module } from '@codesandbox/common/lib/types';
 
-const API_ROOT = '/api/v1';
+export const API_ROOT = '/api/v1';
 
-const getMessage = (error: AxiosError) => {
-  const response = error.response;
-
-  if (!response || response.status >= 500) {
-    logError(error);
-  }
-
-  if (response && response.data) {
-    if (response.data.errors) {
-      const errors = values(response.data.errors)[0];
-      if (Array.isArray(errors)) {
-        if (errors[0]) {
-          error.message = errors[0]; // eslint-disable-line no-param-reassign
-        }
-      } else {
-        error.message = errors; // eslint-disable-line no-param-reassign
-      }
-    } else if (response.data.error) {
-      error.message = response.data.error; // eslint-disable-line no-param-reassign
-    } else if (response.status === 413) {
-      return 'File too large, upload limit is 5MB.';
-    }
-  }
+type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };
 
-  return error.message;
+export type Params = {
+  [key: string]: string;
 };
 
-function handleResponse(
-  response: AxiosResponse,
-  { shouldCamelize = true } = {}
-) {
-  const camelizedData = shouldCamelize
-    ? camelizeKeys(response.data)
-    : response.data;
-
-  // Quickfix to prevent underscored dependencies from being camelized.
-  // Never store data as keys in the future.
-  if (
-    camelizedData &&
-    camelizedData.data &&
-    camelizedData.data.npmDependencies
-  ) {
-    camelizedData.data.npmDependencies = response.data.data.npm_dependencies;
-  }
+export type Options = {
+  shouldCamelize: boolean;
+};
 
-  return camelizedData.data ? camelizedData.data : camelizedData;
-}
+export type Api = {
+  get<T>(path: string, params?: Params, options?: Options): Promise<T>;
```

Comment:
@christianalfoni I think function types are more commonly declared (using the arrow) like

```ts
export type Api = {
  get: <T>(path: string, params?: Params, options?: Options) => Promise<T>;
}
```